frontend/packages/api_client/src/apis/V2Api.ts

```diff
@@ -575,6 +583,52 @@ export class V2Api extends runtime.BaseAPI {
         return await response.value();
     }
 
+    /**
+     * This is a custom handler to partially update theme fields.
+     * Partial update for theme configuration
+     */
+    async instancesOpenedxConfigThemeConfigRaw(requestParameters: InstancesOpenedxConfigThemeConfigRequest): Promise<runtime.ApiResponse<ThemeSchema>> {
+        if (requestParameters.id === null || requestParameters.id === undefined) {
+            throw new runtime.RequiredError('id','Required parameter requestParameters.id was null or undefined when calling instancesOpenedxConfigThemeConfig.');
+        }
+
+        if (requestParameters.data === null || requestParameters.data === undefined) {
+            throw new runtime.RequiredError('data','Required parameter requestParameters.data was null or undefined when calling instancesOpenedxConfigThemeConfig.');
+        }
+
+        const queryParameters: runtime.HTTPQuery = {};
+
+        const headerParameters: runtime.HTTPHeaders = {};
+
+        headerParameters['Content-Type'] = 'application/json';
+
+        if (this.configuration && this.configuration.apiKey) {
+            headerParameters["Authorization"] = this.configuration.apiKey("Authorization"); // api_key authentication
+        }
+
+        if (this.configuration && (this.configuration.username !== undefined || this.configuration.password !== undefined)) {
```

Comment:
@giovannicimolin, shouldn't these 2 if conditions be mutually exclusive and hence this be in an `else if` block?

Reply:
This is auto-generated code, so I didn't even check it. Not sure about these conditions, but I can check.

Reply:
@lgp171188 Can you add a little bit more context here? Is this about the line 609?

Reply:
@giovannicimolin, I was referring to the `if` conditions in the lines 605 and 609.

Reply:
Those lines are like that because there are 2 auth methods defined in our Swagger spec (basic and API key).
I agree with you that they should be mutually exclusive, but I think we don't have control on how that code is generated. 
Since we control the `configuration` variable, and we never set the 2 auths at the same time this should be fine (https://github.com/open-craft/opencraft/blob/new-console/frontend/src/global/api.ts#L3).

Additionally, adding this to the cleanup step of the epic can be done (remove the basic auth from the swagger spec).

Reply:
If we reach a consensus that the generated code here could be better, we could create a backlog item to submit a patch to Swagger in the future.

Reply:
I'm not sure that's a good idea since we don't have any expertise on it (and we also don't have budget). In any case, the current generated code works well and the issue is not really in our usage scope so we're fine to leave this as is for now.